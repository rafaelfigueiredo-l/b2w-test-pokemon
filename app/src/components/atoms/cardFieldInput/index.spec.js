import CardFieldInput from ".";

describe("<CardFieldInput />", () => {
  const onChangeMock = jest.fn();

  const component = (
    <CardFieldInput onChange={onChangeMock()} value="ValueText" />
  );

  const mounted = mount(component);
  const wrapper = shallow(component);

  it("should mount correctly CardFieldInput component", () => {
    expect(wrapper.find(".cardFieldInput")).toHaveLength(1);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly CardFieldInput component", () => {
    expect(mounted.unmount()).toEqual({});
  });

  it("should render input type checkbox", () => {
    expect(wrapper.find("[type='text']")).toHaveLength(1);
  });

  it("should call onChange function", () => {
    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toBeCalled();
  });
});
