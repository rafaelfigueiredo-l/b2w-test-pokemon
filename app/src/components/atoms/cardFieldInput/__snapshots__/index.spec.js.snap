// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardFieldInput /> should mount correctly CardFieldInput component 1`] = `
<styled.input
  className="cardFieldInput"
  onChange={[Function]}
  type="number"
  value="ValueText"
>
  <input
    className="sc-AxjAm dLgPKd cardFieldInput"
    onChange={[Function]}
    type="number"
    value="ValueText"
  />
</styled.input>
`;
